packages/dds/sequence/src/intervals/sequenceInterval.ts

```diff
@@ -239,14 +262,19 @@ export class SequenceInterval implements ISerializableInterval {
 	 * @internal
 	 */
 	public union(b: SequenceInterval) {
+		const newStart = minReferencePosition(this.start, b.start);
+		const newEnd = maxReferencePosition(this.end, b.end);
+
 		return new SequenceInterval(
 			this.client,
-			minReferencePosition(this.start, b.start),
-			maxReferencePosition(this.end, b.end),
+			newStart,
+			newEnd,
 			this.intervalType,
 			undefined,
+			// todo: not happy with how merging here works
 			(this.stickiness | b.stickiness) as IntervalStickiness,
-			this.isExclusive,
+			this.start === newStart ? this.startSide : b.startSide,
```

Comment:
seems like if `this.start === b.start` but they differ on sides, we should prefer `Side.Before` here. similarly for after.